pydm/main_window.py

```diff
@@ -469,6 +470,8 @@ def quit_main_window(self, checked):
         self._confirm_quit(callback=self.app.quit)
 
     def _confirm_quit(self, callback):
+        if not config.CONFIRM_QUIT:
+            return callback()
```

Comment:
You need to execute the callback, not return it.
```suggestion
            callback()
```